rotkehlchen/tests/unit/decoders/test_zerox.py

```diff
@@ -1399,4 +1451,242 @@ def test_swap_settler_token_token_ethereum(ethereum_inquirer, ethereum_accounts)
+        location=Location.BASE,
+        event_type=HistoryEventType.TRADE,
+        event_subtype=HistoryEventSubType.RECEIVE,
+        asset=A_BASE_USDC,
+        amount=FVal(received_amount),
+        location_label=user_address,
+        notes=f'Receive {received_amount} USDC as the result of a swap via the 0x protocol',
+        counterparty=CPT_ZEROX,
+        address=settler_address,
+    )]
+
+
+@pytest.mark.vcr(filter_query_parameters=['apikey'])
+@pytest.mark.parametrize('binance_sc_accounts', [['0xEa7D26292C68033E7535988936637d77495aDa1e']])
+def test_swap_via_settler_on_binance_sc(binance_sc_inquirer, binance_sc_accounts):
+    """Check that a swap via settler on binance_sc is decoded correctly."""
+    tx_hash = deserialize_evm_tx_hash('0x265b58104dba6237018416376611c1d0f170722b19c2949178127f6af496bac7')   # noqa: E501
+    events, _ = get_decoded_events_of_transaction(evm_inquirer=binance_sc_inquirer, tx_hash=tx_hash)  # noqa: E501
+    user_address, timestamp, gas_amount, swap_amount, received_amount, settler_address = binance_sc_accounts[0], TimestampMS(1742451687000), '0.000168127', '1.156691259384', '904876.73552827873791536', '0x4C6F446dD88fD1be8B80D2940806002777dc12a2'  # noqa: E501
+    assert events == [EvmEvent(
+        tx_hash=tx_hash,
+        sequence_index=0,
+        timestamp=timestamp,
+        location=Location.BINANCE_SC,
+        event_type=HistoryEventType.SPEND,
+        event_subtype=HistoryEventSubType.FEE,
+        asset=A_BSC_BNB,
+        amount=FVal(gas_amount),
+        location_label=user_address,
+        notes=f'Burn {gas_amount} BNB for gas',
+        counterparty=CPT_GAS,
+    ), EvmEvent(
+        tx_hash=tx_hash,
+        sequence_index=1,
+        timestamp=timestamp,
+        location=Location.BINANCE_SC,
+        event_type=HistoryEventType.TRADE,
+        event_subtype=HistoryEventSubType.SPEND,
+        asset=A_BSC_BNB,
+        amount=FVal(swap_amount),
+        location_label=user_address,
+        notes=f'Swap {swap_amount} BNB via the 0x protocol',
+        counterparty=CPT_ZEROX,
+        address=settler_address,
+    ), EvmEvent(
+        tx_hash=tx_hash,
+        sequence_index=2,
+        timestamp=timestamp,
+        location=Location.BINANCE_SC,
+        event_type=HistoryEventType.TRADE,
+        event_subtype=HistoryEventSubType.RECEIVE,
+        asset=Asset('eip155:56/erc20:0xB11E6ed6f481fD954Dcd4585d8877aCe06acadfE'),
+        amount=FVal(received_amount),
+        location_label=user_address,
+        notes=f'Receive {received_amount} DFlow as the result of a swap via the 0x protocol',
+        counterparty=CPT_ZEROX,
+        address=settler_address,
+    )]
+
+
+@pytest.mark.vcr(filter_query_parameters=['apikey'])
+@pytest.mark.parametrize('optimism_accounts', [['0xBa41b0e948d681e4dDE13e4B153d89b611e33163']])
+def test_swap_via_settler_on_optimism(optimism_inquirer, optimism_accounts):
+    """Check that a swap via settler on optimism is decoded correctly."""
```

Comment:
I said about removing the comments for the tests since they don't add value. YOu are repeating the test name